STM32F405RGTx/Core/Src/rtos_threads/svpwm_thread.cpp

```diff
@@ -0,0 +1,100 @@
+#include <stdio.h>
+#include <stdint.h>
+#include "main.h"
+#include "threads.hpp"
+#include "math.h"
+
+SVPWM::SVPWM(float SwitchFreq, VHZPROFILE *vhz, TIM_HandleTypeDef *htim){
+	ModIndex = 0;
+	Freq = 0;
+	SwitchFreq = SwitchFreq;
+	Alpha = 0;
+	NewEntry = 0;
+	SectorPointer = 0;
+	U = 0;
+	V = 0;
+	W = 0;
+	StepAngle = 0;
+	EntryOld = 0;
+	ta = 0;
+	tb = 0;
+	to = 0;
+
+	vhz = vhz;
+	htim = htim;
+}
+
+void SVPWM::initialize(){
+   	thread = RTOSThread(
+        "svpwm_thread",
+        1024,
+        osPriorityAboveNormal,
+		update 
+    );
+}
+
+void SVPWM::update(void* arg){

```

Comment:
The update function should contain an infinite while loop containing the code that should run every update. Inside this while loop, you will have at the end an `osDelay(X)` where X is the periodicity at which the thread should update in milliseconds.

Reply:
I've implemented something similar using `osThreadFlagsWait(0x00000001U, osFlagsWaitAll, 0U);`
Where we now only updated once `void HAL_TIM_PeriodElapsedCallback(TIM_HandleTypeDef *htim)` has been called.